atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -250,4 +272,28 @@ public synchronized void shutdown() {
             throw Throwables.rewrapAndThrowUncheckedException(e);
         }
     }
+
+    private enum SweepOutcome {
+        SUCCESS, NOTHING_TO_SWEEP, RETRYING_WITH_SMALLER_BATCH, DISABLED, UNABLE_TO_ACQUIRE_LOCKS,
+        NOT_ENOUGH_DB_NODES_ONLINE, TABLE_DROPPED_WHILE_SWEEPING, ERROR
+    }
+
+    private class SweepOutcomeMetrics {
+        private final MetricsManager metricsManager = new MetricsManager();
+        private SweepOutcome lastOutcome = null;
+
+        SweepOutcomeMetrics() {
+            Arrays.stream(SweepOutcome.values()).forEach(outcome ->
+                    metricsManager.registerMetric(BackgroundSweeperImpl.class, outcome.name(), () -> getCount(outcome))
```

Comment:
Yeah, this was a simplification.  We lose some information by only sampling the last result rather than reporting all results.  This is enough for spotting "is sweep disabled for long periods of time" but bad for finer grained detail.  Is it interesting to know that sweep was disabled for 25 seconds?

Things get more complicated if we want to record counts for the outcomes in the period.  Firstly not sure if we can assume there is only 1 collector running.  If more than 1 is running we can't zero the metrics when the lambda is called.

We could use a `Meter` but then we get weird decay functions on the numbers so we see something of the shape of the outcomes but it is difficult to tie back to actual numbers.  Alternatively we can keep records so we can always report "this is the number of each outcome in the last X minutes".

We should chat tomorrow about goal here.  Whatever we do I think I should add a single logging line for "ran iteration and outcome was X" so the fine grained information is clearly visible in the logs.